### DISCUSSION TITLE
Fill missing data of just one phase

### DISCUSSION THREAD

**Discussion by matheusGariglio** [mtex-toolbox/mtex#1064] [Ask Anything] [answered]
Hi,

I'm working with a two-phase Ti alloy and the beta phase has a much worse indexation compared to the alpha phase as you can see below : 
![image](https://user-images.githubusercontent.com/77332633/119786853-e175c980-bed0-11eb-9979-9af6adf2a701.png)
However, when I try to fill the missing data, there is a tendency of growing the alpha particles, as you can see in my filled data (mostly in the bottom part of the image):
![image](https://user-images.githubusercontent.com/77332633/119787013-0cf8b400-bed1-11eb-9578-613c37978548.png)
I would like to know if there is a way of reconstruction only the beta phase in this kind of ebsd data, as the alpha phase is pretty much well indexed.

If someone is having the same kind of problem indexing the beta phase of a deformed microstructure, I would be glad to know your opinions on how to have a better indexing rate for the deformed beta :)

Cheers,
Matheus

**Comment by kilir** [comment, marked as answer]
Hi,
so assuming you are not interested in the alpha phase, you can set it to 'notIndexed' and just fill the non-existent points.

Above doesn't really work though, however the following might be a workaround:

1) fill everything with B ()
2) place the old A again at the proper positions

The following works with your data:

```

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [3.3 3.3 3.3], 'mineral', 'B', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('6/mmm', [2.9 2.9 4.7], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'A', 'color', [0.56 0.74 0.56])};

ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');

%%
ebsd=ebsd.gridify
ebsd_new=fill(ebsd('B'))
% now find points from ebsd_new which were A and put the old A there again
ebsd_new(ebsd.phase == 2) = ebsd('A')

plot(ebsd_new)
```
![newfilesomething](https://user-images.githubusercontent.com/11697557/119872990-58887d80-bf24-11eb-8405-c4e6119c96da.png)


Cheers,
Rüdiger

**Reply by matheusGariglio** [reply]
Hi Rüdiger,

Thanks a lot again! It worked fine!

Cheers,
Matheus

**Reply by matheusGariglio** [reply]
Hi Rüdiger,

I would like to know if there is a way for doing the same thing that you did here :
```
ebsd=ebsd.gridify
ebsd_new=fill(ebsd('B'))
% now find points from ebsd_new which were A and put the old A there again
ebsd_new(ebsd.phase == 2) = ebsd('A')
```

In order to return some big 'notIndexed' areas, for example :
```
notInd = grains('notIndexed')
toKeep = notInd(notInd.grainSize./notInd.boundarySize > 2.5*mean(notInd.grainSize ./ notInd.boundarySize));

ebsd_toKeep = ebsd(toKeep)
ebsd_new(ebsd_toKeep) = ebsd('notIndexed')
```

I was trying to do this script above, but is not working. I would really appreciate if you could help me on this please :)

Cheers,
Matheus
